execution_engine/src/runtime_context/mod.rs

```diff
@@ -113,36 +121,39 @@ where
     /// Where we already have a runtime context, consider using `new_from_self()`.
     #[allow(clippy::too_many_arguments)]
     pub fn new(
-        tracking_copy: Rc<RefCell<TrackingCopy<R>>>,
-        entry_point_type: EntryPointType,
         named_keys: &'a mut NamedKeys,
-        access_rights: ContextAccessRights,
-        runtime_args: RuntimeArgs,
+        entity: &'a AddressableEntity,
+        entity_address: Key,
```

Comment:
i think this is initiating_entity, and the type should be AddressableEntityHash instead of Key